### PR TITLE
[#166572994] Fix payment ID infinite polling

### PR DESCRIPTION
**Pull request by Undermaken**
**Short description:**
This PR fixes infinite polling on payment id requests

- create a new backend client for every polling request
- each client has its own deferredPromise
- on cancel polling request, backend client will be stopped by using the relative deferredPromise instance

### CODE REVIEW

File: ts/sagas/wallet.ts [hunk 1]
```diff
@@ -636,11 +631,29 @@ export function* watchWalletSaga(
...
-  );
+  yield takeLatest(getType(paymentIdPolling.request), function*(
+    action: ActionType<typeof paymentIdPolling["request"]>
+  ) {
+    shouldAbortPaymentIdPollingRequest = DeferredPromise<boolean>();
+    // Backend client for polling for paymentId - uses an instance of fetch that
+    // considers a 404 as a transient error and retries with a constant delay
+    // it is created every time paymentIdPolling.request is triggered:
+    // this is because we have to re-new shouldAbortPaymentIdPollingRequest to be able to abort the running polling
+    const pollingPagopaNodoClient = BackendClient(
```
Comment (gunzip) [inline ts/sagas/wallet.ts:642]: another option is to create the deffered promise and return it inside the fetch call

File: ts/api/backend.ts [hunk 2]
```diff
@@ -347,9 +350,22 @@ export function BackendClient(
     postAttivaRpt: withBearerToken(
       createFetchRequestForApi(attivaRptT, options)
     ),
-    getPaymentId: withBearerToken(
-      createFetchRequestForApi(getPaymentIdT, options)
-    ),
+    getPaymentId: (maxPollingRetries: number, retryDelay: Millisecond) => {
```
Comment (cloudify) [inline ts/api/backend.ts:353]: `maxPollingRetries` and `retryDelay` don't change between calls to `getPaymentId`, you can move these options to the `BackendClient` constructor

File: ts/utils/fetch.ts [hunk 3]
```diff
@@ -119,5 +120,7 @@ export const constantPollingFetch = (
   );
 
   // TODO: remove the cast once we upgrade to tsc >= 3.1 (https://www.pivotaltracker.com/story/show/170819445)
```
Comment (cloudify) [inline ts/utils/fetch.ts:122]: is this TODO still relevant?

File: ts/sagas/wallet/pagopaApis.ts [hunk 4]
```diff
@@ -661,8 +661,8 @@ export function* paymentIdPollingRequestHandler(
       typeof isPagoPATestEnabledSelector
     > = yield select<GlobalState>(isPagoPATestEnabledSelector);
 
-    const response: SagaCallReturnType<typeof getPaymentIdApi> = yield call(
-      getPaymentIdApi,
+    const response: SagaCallReturnType<typeof getPaymentIdApi.e2> = yield call(
+      getPaymentIdApi.e2,
```
Comment (cloudify) [inline ts/sagas/wallet/pagopaApis.ts:665]: for clarity it would be worth destructuring the duple into temporary variables with proper names

**Pull request merged**